ssl/ssl_lib.cc

```diff
@@ -480,6 +480,20 @@ bool SSL_get_traffic_secrets(const SSL *ssl,
   return true;
 }
 
+void ssl_update_counter(SSL_CTX *ctx, int &counter, bool locked) {
+  if (locked) {
+    MutexWriteLock lock(&ctx->lock);
```

Comment:
This locks a SSL_CTX lock.

Reply:
Yes, this is because the `stats` structure is nested in `SSL_CTX` so we use the lock that was already with it.